### ISSUE TITLE
Specifying custom CUDA path over existing one

### ISSUE THREAD

**Issue by L4in** [torch/cutorch#428]
Hi,

While trying to install Torch7 with CUDA on a shared cluster, I'm met with the following error:

```
Found CUDA on your machine. Installing CUDA packages
Warning: unmatched variable LUALIB
cmake -E make_directory build && cd build && cmake .. -DLUALIB= -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="/home/mclement/torch/install/bin/.." -DCMAKE_INSTALL_PREFIX="/home/mclement/torch/install/lib/luarocks/rocks/cutorch/scm-1" && make -j$(getconf _NPROCESSORS_ONLN) install

-- Found Torch7 in /home/mclement/torch/install
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find CUDA: Found unsuitable version "6.0", but required is at
  least "6.5" (found /usr/local/cuda)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:313 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindCUDA.cmake:806 (find_package_handle_standard_args)
  CMakeLists.txt:7 (FIND_PACKAGE)
```

This is to be expected, as someone else is using version 6.0 that have been linked to `/usr/local/cuda`. Symlinking `/usr/local/cuda-6.5` to `/usr/local/cuda` is not currently an option.

However, even with `CUDA_BIN_PATH`set to `/usr/local/cuda-6.5`, the error stays the same.

How to override the `/usr/share/cmake/Modules/FindCUDA.cmake` pointing on the wrong directory without modifying any shared resource?

Thanks in advance,
C. K. Michel

**Comment by johnPertoft** [comment]
Have you found a solution to this?

**Comment by L4in** [comment]
Unfortunately I still haven't.

**Comment by zhangxiangxiao** [comment]
Try
```bash
export PATH=/usr/local/cuda-6.5/bin:$PATH
```

It seems CUDA is found by testing nvcc. Therefore, `CUDA_BIN_PATH` could mean `/usr/local/cuda-6.5/bin` instead of its parent. I might be wrong.

**Comment by L4in** [comment]
> Therefore, CUDA_BIN_PATH could mean /usr/local/cuda-6.5/bin instead of its parent. I might be wrong.

That's also what I thought, so I tried both with `/usr/local/cuda-6.5` and `/usr/local/cuda-6.5/bin` without success.

I've also tried `CUDA_BIN_PATH` both as pointing to the `bin` and the `cuda-6.5` repositories, also without success.

**Comment by johnPertoft** [comment]
Yep, that didn't work for me either. In fact I removed all other instances of nvcc on the system than the one in the correct cuda installation. I solved the torch installation issue by just bypassing the part where CMAKE tries to find CUDA by itself and specified them myself instead.

**Comment by lakehanne** [comment]
@zhangxiangxiao 's `export PATH=/usr/local/cuda-6.5/bin:$PATH` seems to work for me. I pointed path to cuda-7.5 though.

Cheers!

**Comment by Jeffalltogether** [comment]
I am having the same issue.  I can successfully install caffe2, but pytorch consistently finds the older version of cuda on my system ahead of the correct version.  @johnPertoft could you explain how you bypassed, " the part where CMAKE tries to find CUDA"?

**Comment by ghltshubh** [comment]
cuda-8.0.6 seems to be missing bin directory for some reason.

**Comment by BenjaminLAZARD** [comment]
@Jeffalltogether I believe you have to edit `.../torch/install/share/cmake/torch/FindCUDA/run_nvcc.cmake` and change the path for nvcc (replace the variable which starts by @ by the path towards your cuda9.0 nvcc).
Installation worked for me after this, although I am I am not 100% sure this is the reason why
